src/dask_awkward/lib/core.py

```diff
@@ -85,14 +85,22 @@ def __init__(
         self,
         dsk: HighLevelGraph,
         name: str,
-        meta: Any,
+        meta: Any | None = None,
+        dtype: DTypeLike | None = None,
         known_value: Any | None = None,
     ) -> None:
         if not isinstance(dsk, HighLevelGraph):
             dsk = HighLevelGraph.from_collections(name, dsk, dependencies=())  # type: ignore
         self._dask: HighLevelGraph = dsk
         self._name: str = name
-        self._meta: Any = self._check_meta(meta)
+        if meta is not None and dtype is None:
+            self._meta = self._check_meta(meta)
+            self._dtype = np.dtype(self._meta.type.content.primitive)
```

Comment:
If the meta always has a `NumpyType` type, we could also just use `self._meta.layout.dtype`. I think that might be slightly better, because it uses Awkward's dtype→primitive conversion.